### PR TITLE
added code documentation for MediaNetAdSdk Module

### PR DESCRIPTION
**Pull request by bharathsu1411**
added comments to all classes and methods of MediaNetAdSdk Module

### CODE REVIEW

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt [hunk 1]
```diff
@@ -199,10 +256,23 @@ object MediaNetAdSDK {
         EventManager.clear()
     }
 
+    /**
+     * Gets any given subject to GDPR in that order. <br>
+     * 1) MediaNetAdSdk subject to GDPR custom value, if present. <br>
```
Comment (nikhilAgarwalM) [inline MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt:261]: <br>?

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt [hunk 2]
```diff
@@ -10,12 +10,20 @@ import com.medianet.android.adsdk.events.Constants.EventName.BID_REQUEST
...
+ * Manager Class that handles Events transmission to Analytics SDK
+ */
 object EventManager {
 
     fun init(config: SdkConfiguration) {
         EventFactory.updateConfiguration(config)
     }
 
+    /**
+     * sends Bid Request Event to Analytics when Bid Auction Call is made
```
Comment (nikhilAgarwalM) [inline MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt:23]: "...Bid Request is made"

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt [hunk 3]
```diff
@@ -25,6 +33,11 @@ object EventManager {
         )
     }
 
+    /**
+     * sends Timeout Event to Analytics when Bid Request Auction Call times out
```
Comment (nikhilAgarwalM) [inline MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt:37]: replace auction with request

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt [hunk 4]
```diff
@@ -34,6 +47,11 @@ object EventManager {
         )
     }
 
+    /**
+     * sends event to analytics when adRequest is sent to GAM after the auction
```
Comment (nikhilAgarwalM) [inline MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt:51]: omit auction and replace it with Bid Request or bid request to server

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/nativead/NativeAd.kt [hunk 5]
```diff
@@ -10,21 +10,38 @@ import com.medianet.android.adsdk.utils.Util.getPrebidAssetFromNativeAdAsset
...
+ */
 class NativeAd(adUnitId: String): Ad(NativeAdUnit("imp-prebid-banner-native-styles")) {
     // TODO Pass adUnitId to NativeAdUnit once it is configured
     private var mNativeAdUnit: NativeAdUnit = adUnit as NativeAdUnit
 
     override val adType: AdType = AdType.NATIVE
 
+    /**
+     * sets the context type for the native ad
+     * which will in turn be sent in the bid request for auction call
```
Comment (nikhilAgarwalM) [inline MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/nativead/NativeAd.kt:24]: replace auction with bid request ot bid request to server here and at other places

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/Ad.kt [hunk 6]
```diff
@@ -57,13 +64,37 @@ abstract class Ad(val adUnit: AdUnit) {
...
+     * cancels auto refresh of ad
+     */
     fun stopAutoRefresh() = adUnit.stopAutoRefresh()
+
+    /**
+     * resumes auto refresh of ad
+     */
     fun resumeAutoRefresh() = adUnit.resumeAutoRefresh()
 
+    // MARK: - adunit context data aka inventory data (imp[].ext.context.data)
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/Ad.kt]: not required, removed it.

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/Ad.kt [hunk 7]
```diff
@@ -72,13 +103,21 @@ abstract class Ad(val adUnit: AdUnit) {
...
+     */
     fun sendAdLoadedEvent() {
         EventManager.sendAdLoadedEvent(
             dfpDivId = adUnit.configuration.configId,
             sizes = Util.mapAdSizesToMAdSizes(adUnit.configuration.sizes)
         )
     }
 
+    /**
+     * This method initiates the Bid Auction call
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/Ad.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/BannerAd.kt [hunk 8]
```diff
@@ -13,7 +13,9 @@ import org.prebid.mobile.BannerAdUnit
 import org.prebid.mobile.addendum.AdViewUtils
 import org.prebid.mobile.addendum.PbFindSizeError
 
-
+/**
+ * Banner Ad Class for original type
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/BannerAd.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/BannerAd.kt [hunk 9]
```diff
@@ -23,14 +25,29 @@ class BannerAd(adUnitId: String, val adSize: AdSize = AdSize.BANNER): Ad(BannerA
     override val adType: AdType = AdType.BANNER
 
 
+    /**
+     * this method allows us to add multiple sizes to the ad
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/BannerAd.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/ConfigSerializer.kt [hunk 10]
```diff
@@ -7,6 +7,9 @@ import com.medianet.android.adsdk.model.StoredConfigs.StoredSdkConfig
 import java.io.InputStream
 import java.io.OutputStream
 
+/**
+ * Serializer Class to write in and read from datastore
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/ConfigSerializer.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt [hunk 11]
```diff
@@ -68,14 +78,18 @@ object MediaNetAdSDK {
...
             //TODO - that need to be come from customer
-            TargetingParams.setSubjectToGDPR(true)
+            setSubjectToGDPR(true)
+            publisherSdkInitListener = sdkInitListener
+            initialiseSdkConfig(applicationContext)
         }
     }
 
+    /**
+     * fetches sdk config for account ID
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt [hunk 12]
```diff
@@ -99,19 +113,33 @@ object MediaNetAdSDK {
...
+      */
     private fun initConfigExpiryTimer(applicationContext: Context, expiry: Long?) {
         expiry?.let { expiryInSeconds ->
             CustomLogger.debug(TAG, "refreshing config after $expiryInSeconds seconds")
             SDKConfigSyncWorker.scheduleConfigFetch(applicationContext, expiryInSeconds)
         }
     }
 
+    /**
+     * extracts data from config that will be required for auction call
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt [hunk 13]
```diff
@@ -187,7 +242,9 @@ object MediaNetAdSDK {
         AnalyticsSDK.init(applicationContext, configuration, providers = listOf(analyticsProvider))
     }
 
-    // This method is being used by every Ad class before any functioning
+    /**
+     * This method is being used by every Ad class before any functioning
+      */
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt:249]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt (hunk 2 above)
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt (hunk 3 above)
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt (hunk 4 above)
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/events/EventManager.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/nativead/NativeAd.kt (hunk 5 above)
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/nativead/NativeAd.kt]: done

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt [hunk 14]
```diff
@@ -149,6 +177,10 @@ object MediaNetAdSDK {
         return PrebidMobile.getPbsDebug()
     }
 
+    /**
+     * sets stored auction response
```
[the remark this answers is not in the document]
Comment (bharathsu1411) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt:182]: done

**Review by psiipher — APPROVED** [review]

**Review by nikhilAgarwalM — APPROVED** [review]

**Pull request merged**